matlab/+caffe/private/caffe_.cpp

```diff
@@ -586,6 +586,41 @@ static handler_registry handlers[] = {
 };
 
 /** -----------------------------------------------------------------
+** failure handling.
+**/
+
+// Failure function to prevent Matlab from crushing
+// by throwing an exception
+// Failurefunction might be called multiple times, therefore set a first-call variable
+bool firstFail = true;
+void MyFailureFunction(){
+	if (Caffe::using_Matlab() && firstFail){
+		firstFail = false;
+		mexPrintf("failure function\n");
+		throw ("Fatal Error");
```

Comment:
The idea here is that the error message will be printed to matlab from matlabsink, and the failure function (called at last) aims to jump to caffe_.cpp by throwing an error. Note that previously the failure function is simply abort(), I guess a normal exit in caffe_ might also work (?).